hack/verify.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+projectdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+
+cd "${projectdir}"
+
+tmp="$(mktemp --directory --tmpdir verify.sh.XXXXXXXX)"
+
+cleanup() {
+    rm -rf "${tmp}"
+}
+trap "cleanup" EXIT SIGINT
+
+cp -a "${projectdir}/." "${tmp}"
+pushd "${tmp}" >/dev/null
+
+"$@"
+
+popd >/dev/null
+
+if ! diff --new-file --text --unified --show-c-function --recursive "${projectdir}" "${tmp}"
+then
+    echo
+    echo "Project '${projectdir}' is out of date."
+    echo "Please run '${*}'"
+    exit 1
+fi
```

Comment:
we should probably run this multiple times to make sure our generated output is deterministic

Reply:
Do you mean repeat the verify commands automatically, or do you mean that I should run this a few times locally to ensure it gives repeatable results?

If the latter, I have run it locally multiple times and in the CI environment.
Initially it was giving unpredictable results, because of `go build` operations silently modifying the go.mod files.
But I've set a GOFLAG to prevent any modifications and I've modified the `go get` command in the Makefile to prevent it attempting to change the go.mod file.

Reply:
I was suggesting for former - I've seen bugs introduced in the past where generated files are non-deterministic, causing flakey verification

Reply:
Understood. I'll add the repeat check if we encounter any flakes.